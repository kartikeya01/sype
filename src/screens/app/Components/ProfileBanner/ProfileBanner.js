import React from 'react';
import {View, StyleSheet} from 'react-native';
import ProfileBannerAvatar from '../ProfileBannerAvatar/ProfileBannerAvatar';
import ProfileUsername from '../ProfileUsername/ProfileUsername';
import FollowerButton from '../FollowerButton/FollowerButton';
import FollowingButton from '../FollowingButton/FollowingButton';
import PostCountButton from '../PostCountButton/PostCountButton';

export default class ProfileBanner extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    const {
      //user_id,
      username,
      user_id,
      profile_image,
      following_count,
      follower_count,
      posts_count,
    } = this.props;

    return (
      <View style={styles.profileBannerContainer}>
        <ProfileBannerAvatar profile_image={profile_image} />
        {/** Create three rows --- Possibly 4 but wull get back to this
         * 1. Username
         * 2. Profile Content
         * 3. Profile Display Views
         */}

        <View style={styles.interactiveContentContainer}>
          <View style={styles.rowContainer}>
            <ProfileUsername username={username} />
          </View>

          <View style={styles.rowContainer}>
            <FollowerButton user_id={user_id} follower_count={follower_count} />
            <FollowingButton
              user_id={user_id}
              following_count={following_count}
            />
          </View>

          <View style={styles.rowContainer}>
            <PostCountButton posts_count={posts_count} />
          </View>
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  profileBannerContainer: {
    flexDirection: 'row',
    paddingVertical: 5,
    paddingHorizontal: 5,
  },
  usernameFontSize: {
    fontSize: 25,
  },
  rowContainer: {
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
  },
  interactiveContentContainer: {
    flex: 1,
  },
});
