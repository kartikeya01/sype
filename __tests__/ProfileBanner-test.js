import 'react-native';
import React from 'react';
import ProfileBanner from '@screens/app/Components/ProfileBanner/ProfileBanner';

import renderer from 'react-test-renderer';

test('Profile Banner Snapshot Test', () => {
  const tree = renderer.create(<ProfileBanner />).toJSON();
  expect(tree).toMatchSnapshot();
});
