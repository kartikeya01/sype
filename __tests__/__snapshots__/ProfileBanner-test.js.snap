// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Banner Snapshot Test 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "paddingHorizontal": 5,
      "paddingVertical": 5,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#6200ee",
            "borderRadius": 37.5,
            "height": 75,
            "width": 75,
          },
          undefined,
        ]
      }
    >
      <Image
        source={
          Object {
            "uri": undefined,
          }
        }
        style={
          Object {
            "borderRadius": 37.5,
            "height": 75,
            "width": 75,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 25,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderColor": "white",
            "borderRadius": 16,
            "borderStyle": "solid",
            "borderWidth": 0.5,
            "elevation": 0,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityStates={Array []}
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 16,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "paddingHorizontal": 4,
              }
            }
          >
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "lineHeight": 24,
                      "marginVertical": 4,
                      "minHeight": 24,
                      "textAlignVertical": "center",
                    },
                    Object {
                      "color": "rgba(0, 0, 0, 0.87)",
                      "fontFamily": "System",
                      "fontWeight": "400",
                      "marginLeft": 8,
                      "marginRight": 8,
                    },
                    undefined,
                  ],
                ]
              }
            >
              Followers 
              0
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderColor": "white",
            "borderRadius": 16,
            "borderStyle": "solid",
            "borderWidth": 0.5,
            "elevation": 0,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityStates={Array []}
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 16,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "paddingHorizontal": 4,
              }
            }
          >
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "lineHeight": 24,
                      "marginVertical": 4,
                      "minHeight": 24,
                      "textAlignVertical": "center",
                    },
                    Object {
                      "color": "rgba(0, 0, 0, 0.87)",
                      "fontFamily": "System",
                      "fontWeight": "400",
                      "marginLeft": 8,
                      "marginRight": 8,
                    },
                    undefined,
                  ],
                ]
              }
            >
              Following 
              0
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderColor": "white",
            "borderRadius": 16,
            "borderStyle": "solid",
            "borderWidth": 0.5,
            "elevation": 0,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityStates={Array []}
          accessible={true}
          focusable={false}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 16,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "paddingHorizontal": 4,
              }
            }
          >
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "lineHeight": 24,
                      "marginVertical": 4,
                      "minHeight": 24,
                      "textAlignVertical": "center",
                    },
                    Object {
                      "color": "rgba(0, 0, 0, 0.87)",
                      "fontFamily": "System",
                      "fontWeight": "400",
                      "marginLeft": 8,
                      "marginRight": 8,
                    },
                    undefined,
                  ],
                ]
              }
            >
              Posts 
              NaN
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
